src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -185,6 +185,17 @@ internal NativeCommandProcessor(ApplicationInfo applicationInfo, ExecutionContex
 
             //Create input writer for providing input to the process.
             _inputWriter = new ProcessInputWriter(Command);
+
+            try
+            {
+                Host.BufferCell[,] bufferContents = this.Command.Context.EngineHostInterface.UI.RawUI.GetBufferContents(
+                    new Host.Rectangle(_startPosition, _startPosition));
+                _scrapeHostOutput = true;
+            }
+            catch (NotImplementedException)
```

Comment:
By putting the call to `GetBufferContents` in the constructor, it's possible that an arbitrary exception will be thrown from `GetBufferContents` in the constructor. When that happens, it would be wrapped to an `ApplicationFailedException` properly prior to this change, but after this change, the `NativeCommandProcessor` would just fail to be constructed, and thus I guess the error would look more obscure. So, I'm not sure if it's appropriate to put this in the constructor ...

Reply:
will fix